### DISCUSSION TITLE
Values are out of sync when a field disabled attribute depends of another field value

### DISCUSSION THREAD

**Discussion by olivierpascal** [react-hook-form/react-hook-form#11735] [Q&A]
### Version Number

7.51.2

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/muddy-sky-kcshvj?file=%2Fsrc%2FApp.tsx

### Steps to reproduce

1. Go to https://codesandbox.io/p/sandbox/muddy-sky-kcshvj?file=%2Fsrc%2FApp.tsx

2. Check "Enabled", which will enable "First Name"

values: `{ "isEnabled": true } // OK`

3. Populate "First Name"

values: `{ "isEnabled": true, "firstName": "Bob" } // OK`

4. Uncheck "Enabled"

values: `{ "isEnabled": false, "firstName": "Bob" } // NOT OK`
expected values: `{ "isEnabled": false }`

5. Check again "Enabled"

values: `{ "isEnabled": true } // NOT OK`
expected values: `{ "isEnabled": true, "firstName": "Bob" }`

`isEnabled` and `firstName` are out of sync.

### Expected behaviour

You may argue that this is the expected behavior because `watch` can't predict what I will do next, once `isEnabled` is checked.

I will argue that since the `disabled` attribute change the value, it should trigger a `watch` update.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by olivierpascal** [comment]
Workaround:

```tsx
React.useEffect(() => {
  const subscription = watch(() =>
    setTimeout(() => console.log('values:', getValues()), 0)
  );

  return () => subscription.unsubscribe();
}, [watch, getValues]);
```

**Comment by olivierpascal** [comment]
Could you please explain why this is not a bug, and what is the recommended way to deal with it?

**Reply by swrobel** [reply]
I believe this question is for @bluebill1049
